[WFGP-239] Resettable System Properties for embedded servers

yersan opened the pull request:
Jira issue: https://issues.redhat.com/browse/WFGP-239

Build on top of #244

jfdenise commented on config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/BaseConfigGenerator.java:
```diff
@@ -56,10 +57,52 @@ public abstract class BaseConfigGenerator {
    protected PrintWriter scriptWriter;
    protected StringBuilder scriptBuf;
 
+   protected Set<String> resetProperties = Set.of("java.version",
```
Can't we reverse and only list the system properties to reset? This seems more robust to JVM changes. This set should be shared between feature-spec and config generation.

yersan replied:
That is another option but it would need to know exactly what Sys Prop can affect the embedded server, there are several properties that can be set by the user and can influence the embedded server. Since the setting is out of our control, I added the most restrictive set. 

We could reverse it, but that could be more fragile in the future since any new property that can influence the startup of the embedded server would need to be into account. Another option to complement this is to allow the user to specify with another plugin option, which Sys Prop from the current environment doesn't want to reset, basically, allow the user to extend this resetProperties set.

Do you still prefer to reverse the list? I think I could find the exact Sys Prop but it is more difficult to be sure what other ones can affect...I'm more thinking of any kind of user property that can even influence the XML parsers or factories used by the embedded server

yersan replied:
As an example of the fragility, notice we discovered accidentally one "javax.management.builder.initial"

jfdenise replied:
@yersan , I think that we control wildfly and know the set of system properties that could affect the provisioning. We don't control the JVM system properties. I would vote for reversing and making this property a list of properties a user could extend. Having an empty list would mean to just reset the set we know.

yersan replied:
> We don't control the JVM system properties

We don't change them when we are building WildFly, notice we generally use the default one when we are building the project.

But, ok, I can look at what could be the initial set to remove, although I'm more confident in using a more restrictive set.

yersan replied:
> This set should be shared between feature-spec and config generation.

@jfdenise Right, I did not find the correct place to place it, I'll do another review, do you have any place in mind?

jfdenise replied:
We could have it there: ForkedEmbeddedUtil That is in galleon-plugins, so should be shared by both config and feature-spec.

yersan commented:
@jfdenise I modified the PR. For the sake of completeness, I've added three configurations, one to enable resetting the sys properties and two additional to add or remove system properties from this default set, so in case we need it we could modify it by using these two additional configurations

yersan commented:
@jfdenise As we discussed, simplified to a single property and only use it for the configuration generation phase
